packages/account/src/Components/api-token/api-token-clipboard.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIsMounted } from '@deriv/shared';
+import { Button, Icon, Modal, Text, Popover } from '@deriv/components';
+import { localize } from '@deriv/translations';
+
+const WarningNoteBullet = ({ message }) => (
+    <div className='da-api-token__bullet-wrapper'>
+        <div className='da-api-token__bullet' />
+        <Text as='p' color='prominent ' size='xs' line_height='m'>
+            {message}
+        </Text>
+    </div>
+);
+
+const WarningDialogMessage = () => (
+    <>
+        <Text as='p' color='prominent ' size='xs' line_height='m'>
+            {localize(
+                'Be careful who you share this token with. Anyone with this token can perform the following actions on your account behalf'
+            )}
+        </Text>
+        <div className='da-api-token__bullet-container'>
+            <WarningNoteBullet message={localize('Add accounts')} />
+            <WarningNoteBullet message={localize('Create or delete API tokens for trading and withdrawals')} />
+            <WarningNoteBullet message={localize('Modify account settings')} />
+        </div>
+    </>
+);
+
+const ApiTokenClipboard = ({ scopes, text_copy, info_message, success_message, popoverAlignment = 'bottom' }) => {
+    const [is_copied, setIsCopied] = React.useState(false);
+    const [is_modal_open, setIsModalOpen] = React.useState(false);
+    const [is_popover_open, setIsPopoverOpen] = React.useState(false);
+    const isMounted = useIsMounted();
+    let timeout_clipboard = null;
+    let timeout_clipboard_2 = null;
+    const has_admin_scope = scopes.includes('Admin');
+
+    const toggleModalVisibility = () => {
+        setIsModalOpen(!is_modal_open);
+    };
+
+    const onMouseEnterHandler = () => {
+        if (!is_copied) setIsPopoverOpen(true);
+    };
+
+    const onMouseLeaveHandler = () => {
+        if (!is_copied) setIsPopoverOpen(false);
+    };
+
+    const copyToClipboard = text => {
+        const textField = document.createElement('textarea');
+        textField.innerText = text;
+        document.body.appendChild(textField);
+        textField.select();
+        document.execCommand('copy');
+        textField.remove();
+    };
+
+    const onClick = () => {
+        setIsModalOpen(false);
+        copyToClipboard(text_copy);
+        setIsCopied(true);
+        setIsPopoverOpen(true);
+        timeout_clipboard = setTimeout(() => {
+            if (isMounted()) {
+                setIsPopoverOpen(false);
+            }
+        }, 1900);
+        timeout_clipboard_2 = setTimeout(() => {
+            if (isMounted()) {
+                setIsCopied(false);
+            }
+        }, 2050);
+    };
+
```

Comment:
two timeouts were created for preventing blinking popup message (from 'Token copied' to 'Click here to copy token') 
I couldn’t do it with preventing propagation of event. This bug appears only when popup message is handled outside

Reply:
Please leave a comment for the reason behind adding a setTimeout and the selected value.

Reply:
the comment has been left :ok_hand: